framework_tvb/tvb/adapters/uploaders/bids_importer.py

```diff
@@ -0,0 +1,202 @@
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+
+"""
+.. moduleauthor:: Robert Vincze <robert.vincze@codemart.ro>
+"""
+
+import os
+import json
+import numpy
+
+from tvb.adapters.datatypes.db.region_mapping import RegionVolumeMappingIndex
+from tvb.adapters.datatypes.db.structural import StructuralMRIIndex
+from tvb.adapters.datatypes.db.time_series import TimeSeriesVolumeIndex
+from tvb.adapters.datatypes.db.volume import VolumeIndex
+from tvb.core.adapters.exceptions import LaunchException
+from tvb.core.entities.storage import dao
+from tvb.core.neotraits.forms import TraitUploadField
+from tvb.core.neotraits.uploader_view_model import UploaderViewModel
+from tvb.core.neotraits.view_model import Str
+from tvb.core.adapters.abcuploader import ABCUploaderForm, ABCUploader
+from tvb.datatypes.connectivity import Connectivity
+from tvb.datatypes.graph import CorrelationCoefficients
+from tvb.datatypes.time_series import TimeSeriesRegion
+
+
+class BIDSImporterModel(UploaderViewModel):
+    uploaded = Str(
+        label='BIDS dataset (zip)'
+    )
+
+
+class BIDSImporterForm(ABCUploaderForm):
+
+    def __init__(self):
+        super(BIDSImporterForm, self).__init__()
+
+        self.uploaded = TraitUploadField(BIDSImporterModel.uploaded, '.zip', 'uploaded')
+
+    @staticmethod
+    def get_view_model():
+        return BIDSImporterModel
+
+    @staticmethod
+    def get_upload_information():
+        return {
+            'uploaded': '.zip'
+        }
+
+
+class BIDSImporter(ABCUploader):
+
+    _ui_name = "BIDS Importer"
+    _ui_subsection = "bids_importer"
+    _ui_description = "Import a dataset in BIDS format"
+
+    SUBJECT_PREFIX = "sub"
+
+    NET_TOKEN = "net"
+    SPATIAL_TOKEN = "spatial"
+    TS_TOKEN = "ts"
+
+    def get_form_class(self):
+        return BIDSImporterForm
+
+    def get_output(self):
+        return [VolumeIndex, StructuralMRIIndex, TimeSeriesVolumeIndex, RegionVolumeMappingIndex]
+
+    def launch(self, view_model):
```

Comment:
This method had become extremely long. Can it be divided into smaller (maybe private) methods ?
e.g. _build_connectivity(...)